core/src/toga/window.py

```diff
@@ -812,6 +827,35 @@ def closeable(self) -> bool:
     # End Backwards compatibility
     ######################################################################
 
+    ######################################################################
+    # 2024-07: Backwards compatibility
+    ######################################################################
+    @property
+    def full_screen(self) -> bool:
+        """**DEPRECATED** – Use :any:`Window.state`."""
+        warnings.warn(
+            ("`Window.full_screen` is deprecated. Use `Window.state` instead."),
+            DeprecationWarning,
+        )
+        return bool(self.state == WindowState.FULLSCREEN)
+
+    @full_screen.setter
+    def full_screen(self, is_full_screen: bool) -> None:
+        warnings.warn(
+            ("`Window.full_screen` is deprecated. Use `Window.state` instead."),
+            DeprecationWarning,
+        )
+        if is_full_screen and (self.state != WindowState.FULLSCREEN):
+            self._impl.set_window_state(WindowState.FULLSCREEN)
+        elif not is_full_screen and (self.state == WindowState.FULLSCREEN):
+            self._impl.set_window_state(WindowState.NORMAL)
+        else:
+            return
```

Comment:
This doesn't look *wrong*... but it also seems overly complicated. There shouldnt' be a need to replicate the "if already in fullscreen" logic, as set_window_state() should be handling that.